pshtt/pshtt.py

```diff
@@ -651,106 +728,41 @@ def https_check(endpoint):
+        logging.exception("{}: Unknown exception examining certificate deployment.".format(endpoint.url))
+        utils.debug("{}: Unknown exception examining certificate deployment: {}".format(endpoint.url, err))
 
     try:
-        endpoint.https_cert_chain_len = len(cert_plugin_result.received_certificate_chain)
+        endpoint.https_cert_chain_len = len(cert_plugin_result.certificate_deployments[0].received_certificate_chain)
         if (
                 endpoint.https_self_signed_cert is False and (
-                    len(cert_plugin_result.received_certificate_chain) < 2
+                    endpoint.https_cert_chain_len < 2
                 )
         ):
             # *** TODO check that it is not a bad hostname and that the root cert is trusted before suggesting that it is an intermediate cert issue.
             endpoint.https_missing_intermediate_cert = True
-            if(cert_plugin_result.verified_certificate_chain is None):
+            if cert_plugin_result.verified_certificate_chain is None:
                 logging.warning("{}: Untrusted certificate chain, probably due to missing intermediate certificate.".format(endpoint.url))
                 utils.debug("{}: Only {} certificates in certificate chain received.".format(endpoint.url, cert_plugin_result.received_certificate_chain.__len__()))
             elif(custom_trust is True and public_trust is False):
                 # recheck public trust using custom public trust store with manually added intermediate certificates
                 if(PT_INT_CA_FILE is not None):
                     try:
                         cert_plugin_result = None
-                        command = sslyze.plugins.certificate_info_plugin.CertificateInfoScanCommand(ca_file=PT_INT_CA_FILE)
-                        cert_plugin_result = scanner.run_scan_command(server_info, command)
-                        if(cert_plugin_result.verified_certificate_chain is not None):
+                        scanner = Scanner()
+                        command = ScanCommand.CERTIFICATE_INFO
+                        command_extra_args = {
+                            command: CertificateInfoExtraArguments(custom_ca_file=Path(PT_INT_CA_FILE))
+                        }
+                        scan_request = ServerScanRequest(
+                            server_info=server_info,
+                            scan_commands_extra_arguments=command_extra_args,
+                            scan_commands=[command]
+                        )
+                        scanner.queue_scan(scan_request)
+                        # Consume the generator object and retrieve the first result
+                        scan_result = [x for x in scanner.get_results()][0]
+                        cert_plugin_result = scan_result.scan_commands_results[ScanCommand.CERTIFICATE_INFO]
+                        if(cert_plugin_result.certificate_deployments[0].verified_certificate_chain is not None):
```

Comment:
Another question of how we want to handle multiple certificate deployments here as well.